app/src/main/java/de/mvhs/android/zeiterfassung/TimeListActivity.java

```diff
@@ -24,142 +26,158 @@
-                _LOADER_ID, // ID des Loaders
-                null, // Zusatzdaten
-                this); // Callback-Implementierung (diese Klasse)
-    }
+  @Override
+  public boolean onOptionsItemSelected( MenuItem item ) {
+    if( item.getItemId() == R.id.ExportAction ) {
+      _exporter = new CsvExporter( TimeListActivity.this );
+      _exporter.execute();
 
-    @Override
-    public Loader<Cursor> onCreateLoader(int id, Bundle args) {
-        CursorLoader loader = null;
-
-        switch (id) {
-            case _LOADER_ID:
-                loader = new CursorLoader(
-                        getBaseContext(), // Context
-                        TimelogContract.Timelog.CONTENT_URI, // URI für ContentProvider
-                        _COLUMNS, // Spalten
-                        _ONLY_FINISHED, // Filter (WHERE)
-                        null, // Parameter für Filter
-                        TimelogContract.Timelog.Columns.START + " DESC"); // Soriterung
-                break;
-        }
+      return true;
+    }
 
-        return loader;
+    return super.onOptionsItemSelected( item );
+  }
+
+  public void exportDone() {
+    _exporter = null;
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+
+    getSupportLoaderManager().restartLoader(
+        _LOADER_ID, // ID des Loaders
+        null, // Zusatzdaten
+        this ); // Callback-Implementierung (diese Klasse)
+  }
+
+  @Override
+  public Loader<Cursor> onCreateLoader( int id, Bundle args ) {
+    CursorLoader loader = null;
+
+    switch( id ) {
+      case _LOADER_ID:
+        loader = new CursorLoader(
+            getBaseContext(), // Context
+            TimelogContract.Timelog.CONTENT_URI, // URI für ContentProvider
+            _COLUMNS, // Spalten
+            _ONLY_FINISHED, // Filter (WHERE)
+            null, // Parameter für Filter
+            TimelogContract.Timelog.Columns.START + " DESC" ); // Soriterung
+        break;
     }
 
-    @Override
-    public void onLoadFinished(Loader<Cursor> loader, Cursor data) {
-        switch (loader.getId()) {
-            case _LOADER_ID:
-                _adapter.swapCursor(data);
-                break;
-        }
+    return loader;
+  }
+
+  @Override
+  public void onLoadFinished( Loader<Cursor> loader, Cursor data ) {
+    switch( loader.getId() ) {
+      case _LOADER_ID:
+        _adapter.swapCursor( data );
+        break;
     }
+  }
+
+  @Override
+  public void onLoaderReset( Loader<Cursor> loader ) {
+    switch( loader.getId() ) {
+      case _LOADER_ID:
+        _adapter.swapCursor( null );
+        break;
+    }
+  }
 
-    @Override
-    public void onLoaderReset(Loader<Cursor> loader) {
-        switch (loader.getId()) {
-            case _LOADER_ID:
-                _adapter.swapCursor(null);
-                break;
-        }
+  @Override
+  public void onDialogNegativeClick( DialogFragment dialog ) {
+    if( _exporter != null ) {
+      _exporter.cancel( false );
```

Comment:
Wenn das Interface für das Abbrechen in dem `CsvExporter` implementiert wäre, würde man auch keine globale Variable für den Exporter vorhalten müssen ;-)